### PR TITLE
move release/deploy machinery to main pom

### PR DESCRIPTION
**Pull request by jrmartino**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -52,6 +55,40 @@
     <plugin.maven.enforcer.version>3.0.0</plugin.maven.enforcer.version>
   </properties>
 
+  <profiles>
+    <profile>
+      <id>release</id>
```
Comment (bbranan) [inline pom.xml:60]: I'd suggest adding a note in the documentation to call out the need to activate the release profile when performing a release (`mvn deploy -P release`)

File: pom.xml [hunk 2]
```diff
@@ -52,6 +55,40 @@
...
+      <build>
+        <pluginManagement>
+          <plugins>
```
Comment (bbranan) [inline pom.xml:63]: Looking at the requirements to release to maven central: https://central.sonatype.org/publish/requirements/, it points to a need for javadocs and source jars to be included. Consider adding the `maven-source-plugin` and `maven-javadoc-plugin` to this profile.

Comment (bbranan) [reply pom.xml:63]: There may be other requirements from central at that link that'll be needed. It looks like a `<description>` (usually next to `<name>` and `<url>`) is required as well.

File: pom.xml [hunk 3]
```diff
@@ -135,9 +195,25 @@
...
+      </plugin>
+      <plugin>
+        <groupId>org.sonatype.plugins</groupId>
+        <artifactId>nexus-staging-maven-plugin</artifactId>
+      </plugin>
     </plugins>
   </build>
 
+  <distributionManagement>
+    <snapshotRepository>
```
Comment (bbranan) [inline pom.xml:211]: Is there a reason to not also include the release repository here? That will be needed for performing releases.

**Review by bbranan — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (jrmartino) [reply pom.xml:80]: the javadoc and source plugins have been added, as well as a description

File: pom.xml (hunk 3 above)
Comment (jrmartino) [reply pom.xml]: not sure here, i'll look into it. neither of the exemplars (e.g. https://bitbucket.org/simpligility/ossrh-pipeline-demo/src/master/pom.xml ) mentioned a release repository in this section - i wonder if the release repo is understood in the nexus sonatype plugin

File: pom.xml (hunk 1 above)
Comment (jrmartino) [reply pom.xml:77]: do we need a new documentation section for this parent project?

Comment (jrmartino) [reply pom.xml:77]: i have added a parent pom documentation page under dev including this and other information. we should remember to update this page with any important changes we make to the pom

File: pom.xml (hunk 3 above)
Comment (jrmartino) [reply pom.xml]: i haven't been able to track this down, but the actual release is technically outside the scope of this card (SNAPSHOT deploys are covered). i think we can put this off until it is needed - it is an easy add in the configuration.

**Pull request merged**